lib/Extension/LanguageServerIndexer/Handler/IndexerHandler.php

```diff
@@ -92,13 +91,17 @@ public function indexer(CancellationToken $cancel): Promise
 
                 if ($index % 500 === 0) {
                     $usage = MemoryUsage::create();
-                    $this->clientApi->window()->showMessage()->info(sprintf(
-                        'Indexed %s/%s (%s%%) %s',
-                        $index,
-                        $size,
-                        number_format($index / $size * 100, 2),
-                        $usage->memoryUsageFormatted()
-                    ));
+                    $this->clientApi->workDoneProgress()->report(
+                        $token,
+                        sprintf(
+                            '%s/%s (%s%%) %s',
```

Comment:
Just a little nitpick:
```suggestion
                            '%s/%s (%s%%, %s)',
```